src/test/java/seedu/address/testutil/PersonBuilder.java

```diff
@@ -101,7 +108,40 @@ public PersonBuilder withEmail(String email) {
         return this;
     }
 
+    /**
+     * Sets the {@code NokName} of the {@code EditPersonDescriptor} that we are building.
+     */
+    public PersonBuilder withNokName(String name) {
+        this.nokName = new Name(name);
+        return this;
+    }
+
+    /**
+     * Sets the {@code NokPhone} of the {@code EditPersonDescriptor} that we are building.
+     */
+    public PersonBuilder withNokPhone(String phone) {
+        this.nokPhone = new Phone(phone);
+        return this;
+    }
+
+    /**
+     * Sets the {@code NokEmail} of the {@code EditPersonDescriptor} that we are building.
+     */
+    public PersonBuilder withNokEmail(String email) {
+        this.nokEmail = new Email(email);
+        return this;
+    }
+
+    /**
+     * Sets the {@code NokAddress} of the {@code EditPersonDescriptor} that we are building.
+     */
+    public PersonBuilder withNokAddress(String address) {
+        this.nokAddress = new Address(address);
+        return this;
+    }
+
     public Student build() {
+        Nok nok = new Nok(nokName, nokPhone, nokEmail, nokAddress);
```

Comment:
I like this way of building the nok only when you run `build()`, it is more consistent. Good job!